GVRf/Framework/jni/gl/gl_texture.h

```diff
@@ -56,8 +56,10 @@ class GLTexture : public GLPendingTask {
         memcpy(texture_parameters_, texture_parameters, sizeof(int) * 5);
     }
 
-    ~GLTexture() {
-        deleter_->queueTexture(id_);
+    virtual ~GLTexture() {
+        if (GL_TASK_NONE == pending_gl_task_) {
```

Comment:
You can just check against 0. 0 is never a valid texture id. Checking GL_TASK_NONE to mean the task is done is dangerous. There are too many complications, 1) the task could have never started. 2) the task started and finished but ID is invalid.